### PR TITLE
Fyst 462 implement some form d-400 ctc lines for both pdf and xml

### PR DESCRIPTION
**Pull request by jenny-heath**
## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/FYST-462
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!
## What was done?
- Adds some lines to the calculator, xml, and pdf for form d-400
- If you want to review by commit, I basically kept everything for each line in one commit (with the exception of the last one where I added stuff to line_data.yml because I had forgotten about it
## How to test?
- Calculator tests actual different tax cases
- Xml tests for presence of node, mocking calc method
- Pdf tests for filled in field

### CODE REVIEW

File: app/lib/pdf_filler/nc_d400_pdf.rb [hunk 1]
```diff
@@ -54,7 +54,8 @@ def hash_for_pdf
         y_d400wf_li20a_pg2_good: @xml_document.at('IncTaxWith')&.text,
         y_d400wf_li20b_pg2_good: @xml_document.at('IncTaxWithSpouse')&.text,
         y_d400wf_li23_pg2_good: @xml_document.at('NCTaxPaid')&.text,
-        y_d400wf_li25_pg2_good: @xml_document.at('RemainingPayment')&.text
+        y_d400wf_li25_pg2_good: @xml_document.at('RemainingPayment')&.text,
+        y_d400wf_li10a_good: @xml_document.at('NumChildrenAllowed')&.text
```
Comment (arinchoi03) [inline app/lib/pdf_filler/nc_d400_pdf.rb]: why are all these suffixed with `_good`?

File: app/models/direct_file_data.rb [hunk 2]
```diff
@@ -604,6 +605,10 @@ def interest_reported_amount=(value)
     write_df_xml_value(__method__, value)
   end
 
+  def qualifying_children_under_age_ssn_count=(value)
+    write_df_xml_value(__method__, value)
+  end
+
```
Comment (arinchoi03) [inline app/models/direct_file_data.rb:616]: how do you decide when to create a writer for a given field?

File: spec/factories/state_file_nc_intakes.rb [hunk 3]
```diff
@@ -112,7 +112,7 @@
     end
 
     trait :head_of_household do
-      raw_direct_file_data { StateFile::XmlReturnSampleService.new.read('shiloh_mfs') }
+      raw_direct_file_data { StateFile::XmlReturnSampleService.new.read('shiloh_hoh') }
```
Comment (arinchoi03) [inline spec/factories/state_file_nc_intakes.rb:116]: this was just typo in the past?

File: spec/lib/efile/nc/d400_calculator_spec.rb [hunk 4]
```diff
@@ -9,6 +9,55 @@
...
+          let(:intake) { create(:state_file_nc_intake, filing_status: filing_status, raw_direct_file_data: StateFile::XmlReturnSampleService.new.read("nc_shiloh_hoh")) }
+          let(:calculator_instance) { described_class.new(year: MultiTenantService.statefile.current_tax_year, intake: intake) }
+
+          agis_to_deductions.each do |fagi, deduction_amount|
+            it "returns the value corresponding to #{fagi} FAGI multiplied by number of QCs" do
+              intake.direct_file_data.fed_agi = fagi
+              intake.direct_file_data.qualifying_children_under_age_ssn_count = 2
+
+              calculator_instance.calculate
+              expect(calculator_instance.lines[:NCD400_LINE_10B].value).to eq(deduction_amount)
```
Comment (arinchoi03) [inline spec/lib/efile/nc/d400_calculator_spec.rb:53]: [Dust] can we rename the `deduction_amount` to `deduction_amt_for_two_children` to be more explicit about the amount -- I was initially confused that there wasn't `x 2` math going on but realized you had doubled the deduction amounts.

File: app/lib/efile/nc/d400_calculator.rb [hunk 5]
```diff
@@ -33,6 +35,23 @@ def calculate_line_10b
...
+        qualifying_widow: 25500,
+        single: 12750,
+      }.freeze
+      def calculate_line_11
+        STANDARD_DEDUCTIONS[@intake.filing_status]
+      end
+
+      def calculate_line_12a
+        # Add Lines 9, 10b, and 11
+        # line 9 DeductionsFromFAGI is blank
```
Comment (arinchoi03) [inline app/lib/efile/nc/d400_calculator.rb:51]: [why is line 9 blank? I see some notes from the jira story and looks like it wasn't included in the story but not sure what you're looking at to see that it was not included.](https://codeforamerica.atlassian.net/browse/FYST-371?focusedCommentId=11580)

File: app/lib/pdf_filler/nc_d400_pdf.rb (hunk 1 above)
Comment (jenny-heath) [reply app/lib/pdf_filler/nc_d400_pdf.rb]: yes it is provided to us and therefore i have no idea why `_good` 😅

**Review by arinchoi03 — APPROVED** [review]
Looks good so far, thank you for breaking out the features into discrete commits, this will be a helpful resource when I ever get to this type of work!

File: app/models/direct_file_data.rb (hunk 2 above)
Comment (jenny-heath) [reply app/models/direct_file_data.rb:616]: in this case i made it so i could use it in a test 😬

File: spec/factories/state_file_nc_intakes.rb (hunk 3 above)
Comment (jenny-heath) [reply spec/factories/state_file_nc_intakes.rb:116]: indeed! led me down a long and windy road of WHY DOESN'T MY TEST WORK

File: app/lib/efile/nc/d400_calculator.rb (hunk 5 above)
Comment (jenny-heath) [reply app/lib/efile/nc/d400_calculator.rb:51]: in those jira comments it links to [this task](https://codeforamerica.atlassian.net/browse/FYST-470) which says "line 9 is blank"—that's what i was looking at

**Pull request merged**